tests/testsuite.at

```diff
@@ -203,6 +203,49 @@ m4_define([NEGATIVE_TEST],[
   ])
 ])
 
+m4_define([SWIG_LANGUAGES], [[java],
+  [python]])
+
+dnl Positive testcase for SWIG given configuration
+dnl $1 test name
+dnl $2 category
+dnl $3 language
+m4_define([TEST_EVAL_SWIG],[
+  m4_define([TESTNAME],[$1])
+  m4_define([CATEGORY],[$2])
+  m4_define([LANGUAGE],[$3])
+  m4_define([SCRIPTDIR],["$TESTS"/CATEGORY/LANGUAGE])
+  m4_define([TESTDIR],[SCRIPTDIR/TESTNAME])
+  m4_define([PROGRAM],[TESTDIR/TESTNAME.dl])
+  m4_define([FACTS],[TESTDIR/facts])
+  m4_define([EXPECTEDDIR],[TESTDIR/expected])
+
+  # clean files before testing
+  m4_esyscmd_s(.SCRIPTDIR/clean-test-files-LANGUAGE.sh CATEGORY/LANGUAGE/TESTNAME)
+
+  AT_CHECK("$SOUFFLE" -D. -F FACTS PROGRAM)
+  mkdir -p EXPECTEDDIR
+  mv *.csv EXPECTEDDIR
+
+  # invoke test script
+  m4_esyscmd_s(.SCRIPTDIR/souffle-LANGUAGE-gen.sh CATEGORY/LANGUAGE/TESTNAME/TESTNAME.dl)
+
+  cd TESTDIR
+  SORTED_SAME_FILES([*.csv],[EXPECTEDDIR])
+])
+
+dnl Positive testcase for SWIG
+dnl $1 test name
+dnl $2 category
+m4_define([POSITIVE_SWIG_TEST],[
+  m4_foreach([LANGUAGE],[SWIG_LANGUAGES],[
+    AT_SETUP([$1 LANGUAGE])
+    TEST_EVAL_SWIG([$1],[$2],LANGUAGE)
+    AT_CLEANUP([])
```

Comment:
Use this instead of your cleanup script.

Reply:
We used the cleanup script to clean the files which were previously generated. I believe AT_CLEANUP just ends the current test group of which would be POSITIVE_SWIG_TEST in this case.